src/middleware/password-recovery/validate-id-data.ts

```diff
@@ -0,0 +1,55 @@
+import { NextFunction, Response, Request } from "express";
+import { PasswordRecoveryModel } from "../../models/password-recovery/password-recovery.model";
+import {
+  requestIsClaimed,
+  requestIsExpired,
+} from "../../models/password-recovery/utils";
+import { decrypt } from "../../utils/crypto";
+
+/**
+ * This function is called when user initially clicks on e-mail link.
+ * We hit the recovery request db and make sure there is a valid request.
+ * If there isn't, send an error, causing client to show error message
+ * If there is. Send OK and client will see a UI to update their password
+ * @param req
+ * @param res
+ * @param next
+ */
+export async function validateIdDataRequest(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const { id, data } = req.query;
+
+  if (req.query.devBypass && !!req.query.devBypass === true) {
```

Comment:
Maybe we should check that we're not in production here with `const isProduction = process.env.NODE_ENV && process.env.NODE_ENV.match("production");`

`(req.query.devBypass && !!req.query.devBypass === true)` seems a little redundant. `if (value)` already infers if it's any value other then falsy (""||0||false||undefined||null) then it's truthy. And !!req.query.devBypass casts the truthy or falsy to a hard true or false value.
Normally if we're wanting to check that a particular value is set as true on the query we could do a strict comparison between strings, value === "true", or if we only care if the value is not a falsy just the value it's self is sufficient.

Reply:
Hey. Actually I actually deleted this code (but forgot to push my changes). 
This code was just a workaround to get the sever to respond a certain way so I could get the front end UI to render.